internal/provider/resource_vault_cluster.go

```diff
@@ -276,6 +276,33 @@ If a project is not configured in the HCP Provider config block, the oldest proj
 					},
 				},
 			},
+			"vault_plugin": {
+				Description: "The external plugins that are to be installed on the vault cluster",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"plugin_name": {
+							Description: "The name of the plugin - Valid options for plugin name - 'venafi-pki-backend'",
```

Comment:
Question re: the comment in PR description about not validating for `plugin_name` because it would depend on a flag - what's the behavior here if we pass an invalid option?

Is there validation in cloud-vault-service API for the valid plugin names based on the LD flag? And then the expected behavior when you pass an invalid one here to just fail and HCP TFP would return a validation error from the API?

Reply:
the api request would fail on the  cloud-vault-service API , which would cause the terraform here to fail.

Fast failing would be really nice. Thinking we can use the PluginStatus API on vault-service to get valid plugin names. going to look into this tomorrow 

thanks for calling out on the behavior here

Reply:
looking closer it seems tough to make network calls in the validation functions here

Reply:
ooh yeah FWIW I think it's fine to rely on the API to fail (and just let terraform fail), was just double checking that's actually what happens here. Prob not necessary to add a validation function here to call the API - if it fails at the cloud-vault-service API I think it will be quick enough already 😄 

Relying on cloud-vault-service API to fail is also how some of the validation for observability configs works - e.g. in [this customer escalation](https://hashicorp.atlassian.net/browse/HVT-4116?focusedCommentId=277162), and I _think_ historically it's also been how gating on Azure worked during Azure beta/GA/Plus GA since we only ever added those checks in the API and not here (from looking at commit history).

Reply:
Ah I see your point now, thanks for clarifying. You can't rely on the create and update APIs for validation in the same way the other fields do.

Simple but maybe hacky idea: could you add a validation function here that just validates based on a hard-coded list (similar to how we check whether enum values are valid)? And every time we add more plugins we would update that?

Reply:
I was thinking about that, but would really prefer not to maintain a hard-coded list, when it is available over API. I'll try to dig a bit more to see if the API call is possible